api/app/utils/cffdrs.py

```diff
@@ -0,0 +1,122 @@
+""" This module contains functions for computing fire weather metrics.
+"""
+import logging
+import app.utils.r_importer
+from app.utils.singleton import Singleton
+
+
+logger = logging.getLogger(__name__)
+
+
+@Singleton
+class CFFDRS():
+    """ Singleton that loads CFFDRS R lib once in memory for reuse."""
+
+    def __init__(self):
+        # pylint: disable=too-many-function-args
+        self.cffdrs = app.utils.r_importer.import_cffsdrs()
+
+
+class CFFDRSException(Exception):
+    """ CFFDRS contextual exception """
+
+#   From cffdrs R package comments:
+#   FUELTYPE: The Fire Behaviour Prediction FuelType
+#        ISI: Initial Spread Index
+#        BUI: Buildup Index
+#        FMC: Foliar Moisture Content
+#        SFC: Surface Fuel Consumption (kg/m^2)
+#         PC: Percent Conifer (%)
+#        PDF: Percent Dead Balsam Fir (%)
+#         CC: Constant (crown closure)
+#        CBH: Crown to base height(m)
+
+# Returns:
+#   ROS: Rate of spread (m/min)
+#
+
+
+# Computable: SFC, FMC
+# To store in DB: PC, PDF, CC, CBH (attached to fuel type, red book)
+PARAMS_ERROR_MESSAGE = "One or more params passed to R call is None."
+
+DEFAULT_METRICS = {"PC": 1, "PDF": 1, "CC": 1, "CBH": 1}
+
+# TODO: Once PC, PDF, CC, CDH are confirmed, replace these defaults
+# and potentially store then in the DB as columns in FuelType
```

Comment:
are we still waiting to get these values? and how much of an effect do they have on ROS? (i.e., what I'm asking is whether the ROS we calculate and display in HFI calc is completely off because we're only using these default metrics)

Reply:
Still waiting. No response from SMEs yet most likely because they're literally fighting fires.